src/utils.cpp

```diff
@@ -144,6 +144,27 @@ std::string ToCIFS(std::string& UNCName)
   return CIFSname;
 }
 
+bool InsertUser(std::string& UNCName)
+{
+  if (g_szUser.empty())
+    return false;
+
+  if (UNCName.find("smb://") == 0)
+  {
+    std::string SMBPrefix;
+
+    if (!g_szPass.empty())
+      SMBPrefix = "smb://" + g_szUser + ":" + g_szPass + "@";
+    else
+      SMBPrefix = "smb://" + g_szUser + "@";
```

Comment:
What do you think about this small optimization:

<pre>
std::string SMBPrefix = "smb://" + g_szUser;

if (!g_szPass.empty())
   SMBPrefix.append(":" + g_szPass);

SMBPrefix.append("@"); 
</pre>

Reply:
Just an idea, nothing that needs to be changed.

Reply:
Happy to change to that. I have a tendency to make things very explicit from time to time, and this leads to often far to verbose code. Ill try to tidy things up further before any future PR's.